### DISCUSSION TITLE
Settings access (wrong password)

### DISCUSSION THREAD

**Discussion by MBA78** [revk/xESP32-Faikout#273] [Q&A]
I have configured my Faikin correctly but I am not able to access to settings parameters. It seems that my password is not recongnized.
How can I reset it or the module itself ?

Thanks

**Comment by revk** [comment]
LOL.

1. Using esptool and USB lead or serial lead you can clear the necessary part of the flash, or erase it all and reflash.

2. There is a factory reset MQTT command which should work with no password. `command/Faikin/factory` with a payload that is MAC address with `Faikin` on the end.

Either should get you totally reset.

Now, this raises the question of whether that factory reset command should still exist when a password is set?!

**Comment by MBA78** [comment]
Thanks for your quick answer.

I think that I can not use the MQTT command because the configuration is not correctly done in my Faikin module.
Regarding the Esptool, how can I connect the Faikin module to my computer ? Which card I need to purchase ?

**Reply by revk** [reply]
The back of the Faikin has two options.

- A 5 pin header for serial, designed to work with https://www.amazon.co.uk/dp/B0C15K9ZC4
- A 4 pin header designed to wire direct to a USB plug, which is usually easy to make manually.

**Comment by revk** [comment]
P.S. Given MQTT can be over TLS and can have username and password, I am less concerned over the `factory` command existing.

**Comment by MBA78** [comment]
I have ordered the Tasmotizer but it will arrive in one week (minimum). I am located in France.
In the mean time I would mlike to test the "4 pin to USB" solution. By 4 pin you mean "G + - 5" or "T R + 0 G" pins that are available on the back of the module ? How to wire such pins to the USB side ?

![Capture d’écran 2024-03-22 144935](https://github.com/revk/ESP32-Faikin/assets/114833354/f6c33856-2b7c-40df-bcd9-4f714b412036)

Thanks.

**Reply by revk** [reply]
Err, `G` `+` `-` `5` is 4 pins, and `T` `R` `+` `0` `G` is 5 pins.

The USB is the `G`(GND) `+`(DATA+) `-`(DATA-) and `5`(+5V).

**Comment by MBA78** [comment]
Ok Thanks it is clear.
In order to reset the password what is the procedure ? Do I need a special software like "EspTool" ? what are the commands ?

**Comment by MBA78** [comment]
By using EspTool do I need to flash [Faikin-S3-MINI-N4-R2.bin](https://github.com/revk/ESP32-Faikin/blob/main/ESP32/betarelease/Faikin-S3-MINI-N4-R2.bin) file or [Faikin-S3-MINI-N4-R2-bootloader.bin](https://github.com/revk/ESP32-Faikin/blob/main/ESP32/betarelease/Faikin-S3-MINI-N4-R2-bootloader.bin) file ?

**Reply by revk** [reply]
One of the instructions pages has details of the esptool command. But you may be able to just erase some of the flash.

**Reply by MBA78** [reply]
ok I found it.
Is this command correct for the ESP32-S3-MINI-1 V1.1:
esptool.py -p COM3 write_flash 0x0 Faikin-S3-MINI-N4-R2-bootloader.bin 0x8000 partition-table.bin 0xd000 ota_data_initial.bin 0x10000 Faikin-S3-MINI-N4-R2.bin

**Reply by MBA78** [reply]
Sould I use betarelease or release ?

Thanks again.

**Reply by revk** [reply]
Either, but you want `idf.py erase-flash` first to ensure your settings (and password) is zapped.

**Comment by MBA78** [comment]
Thank- you so much. It is working now.

Reagrds.

**Comment by revk** [comment]
Well done.
